### PR TITLE
Backdeploy task_wait_future fix to Swift 5.6

### PR DESCRIPTION
**Pull request by etcwilde**
This PR pulls in enough of the Swift 5.6 standard library to build and run `task_future_wait` and `task_future_wait_throwing` via the backdeploy library, and applies the fix in ef80a315f810419e8278af2aac1bf2bcb8fb0011 (https://github.com/apple/swift/pull/59377).

I've broken the patch into a couple pieces.
 - The first two commits pull in the original swift 5.6 stdlib and shouldn't be too exciting, it's mostly just copy-paste.
 - The third commit is where the fix is actually applied. Note that the `AsyncTask::PrivateStorage::_status()` method didn't exist in 5.6, so I used what appears to be the equivalent `AsyncTask::PrivateStorage::Status` object.

I've checked that this runs by running `build/Ninja-ReleaseAssert/swift-macosx-arm64/bin/swiftc -sdk $(xcrun --show-sdk-path -sdk macosx) -runtime-compatibility-version 5.6 test.swift`. 

test.swift contains:
```swift
func fifteen() async -> Int {
    return 15
}

func twelve() async throws -> Int {
    return 12
}

async let Fifteen = fifteen()
async let Twelve = twelve()
print(await Fifteen)
print(try await Twelve)
```

rdar://95132657

### CODE REVIEW

**Comment by etcwilde** [comment]
@swift-ci please test

File: stdlib/toolchain/Compatibility56/CMakeLists.txt [hunk 1]
```diff
@@ -5,15 +5,20 @@ include_directories("include/" "${SWIFT_STDLIB_SOURCE_DIR}")
...
+  Concurrency/ThreadSanitizer.cpp
+  Concurrency/TaskAlloc.cpp
+  Concurrency/MutexPThread.cpp
 
   TARGET_SDKS ${SWIFT_DARWIN_PLATFORMS}
 
-  C_COMPILE_FLAGS ${CXX_COMPILE_FLAGS}
+  C_COMPILE_FLAGS ${CXX_COMPILE_FLAGS} "-D__STDC_WANT_LIB_EXT1__=1"
   LINK_FLAGS ${CXX_LINK_FLAGS}
-  INCORPORATE_OBJECT_LIBRARIES swiftCompatibilityThreading
```
Comment (etcwilde) [inline stdlib/toolchain/Compatibility56/CMakeLists.txt:16]: This should stay `swiftCompatibilityThreading`

**Comment by etcwilde** [comment]
@swift-ci please test

**Review by kavon — APPROVED** [review]
Looks good to me!

**Comment by etcwilde** [comment]
@swift-ci please test

**Comment by etcwilde** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by etcwilde** [comment]
@swift-ci please test macOS platform

**Comment by etcwilde** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by etcwilde** [comment]
@swift-ci please test macOS

**Comment by etcwilde** [comment]
@swift-ci please test macOS

**Comment by etcwilde** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by etcwilde** [comment]
@swift-ci please test linux

**Comment by etcwilde** [comment]
@swift-ci please test windows

**Comment by etcwilde** [comment]
@swift-ci please test

**Comment by etcwilde** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Pull request merged**